doc/js_tutorials/js_setup/js_setup/js_setup.markdown

```diff
@@ -128,16 +128,20 @@ Building OpenCV.js from Source
 
     For example:
     @code{.bash}
-    emcmake python ./opencv/platforms/js/build_js.py build_js --build_test
+    emcmake python ./opencv/platforms/js/build_js.py build_js --build_test --cmake_option="-DOPENCV_TEST_DATA_PATH=./opencv_extra/testdata"
     @endcode
 
+    @note
+    It requires the [opencv_extra repository](https://github.com/opencv/opencv.git) located in your development environment.
```

Comment:
Lets keep `OPENCV_TEST_DATA_PATH` optional.

Prefer to use environment variable for `OPENCV_TEST_DATA_PATH`.

Reply:
Did i get you right, that the tests for the `QRCodeDetector` should be skipped when the enviroment variable `OPENCV_TEST_DATA_PATH` is not set? Because the tests depend on two files from the `opencv_extra` repository.

Reply:
Right

Reply:
Please drop `OPENCV_TEST_DATA_PATH` changes from the patch